### PR TITLE
Update doc

### PR DESCRIPTION
**Pull request by samarth9008**
Update and rename doc as per

https://app.asana.com/0/0/1206630536448958/1206662112591963/f
https://github.com/sorrentum/sorrentum/issues/665#issuecomment-1960657871

### CODE REVIEW

File: docs/work_tools/data605.basic_development.how_to_guide.md [hunk 1]
```diff
@@ -30,6 +30,12 @@ information flow and make the contribution process seamless.
 ### Supporting OS
 
 - We support Mac x86, Apple Silicon and Linux Ubuntu
+- Install `docker-cli` in MAC using
```
Comment (gpsaggese) [inline docs/work_tools/data605.basic_development.how_to_guide.md:33]: -> On Mac

File: docs/work_tools/data605.basic_development.how_to_guide.md [hunk 2]
```diff
@@ -30,6 +30,12 @@ information flow and make the contribution process seamless.
...
+  ```bash
+  brew install docker
```
Comment (gpsaggese) [inline docs/work_tools/data605.basic_development.how_to_guide.md:35]: > brew ...
...

File: docs/work_tools/sorrentum.development.how_to_guide.md [hunk 3]
```diff
@@ -11,6 +13,12 @@
 # Supporting OS
 
 - We support Mac x86, Apple Silicon and Linux Ubuntu
```
Comment (gpsaggese) [inline docs/work_tools/sorrentum.development.how_to_guide.md:15]: Any chance we can factor out the documentation and say 

```
Follow the docs/work_tools/data605.basic_development.how_to_guide.md and then ...
```

We don't want to maintain too many files in sync

Comment (samarth9008) [reply docs/work_tools/sorrentum.development.how_to_guide.md]: Separated both of them with differences. Only things kept in sorrentum doc is about docker container, jupyter nb and warm-up issue

**Pull request merged**